cloud/awsprovider.go

```diff
@@ -561,3 +619,112 @@ func (*AWS) QuerySQL(query string) ([]byte, error) {
 	return nil, fmt.Errorf("Error getting query results : %s", *qrop.QueryExecution.Status.State)
 
 }
+
+type spotInfo struct {
+	Timestamp   string `csv:"Timestamp"`
+	UsageType   string `csv:"UsageType"`
+	Operation   string `csv:"Operation`
+	InstanceID  string `csv:"InstanceID`
+	MyBidID     string `csv:"MyBidID"`
+	MyMaxPrice  string `csv:"MyMaxPrice"`
+	MarketPrice string `csv:"MarketPrice"`
+	Charge      string `csv:"Charge"`
+	Version     string `csv:"Version"`
+}
+
+func parseSpotData(bucket string, prefix string, projectID string, region string, accessKeyID string, accessKeySecret string) (map[string]*spotInfo, error) {
+	log.Print("bucket " + bucket)
+	log.Print("prefix " + prefix)
+	log.Print("pID " + projectID)
+	log.Print("region " + region)
+	log.Print("accessKeyName " + accessKeyID)
+	log.Print("secret " + accessKeySecret)
+
+	if accessKeyID != "" && accessKeySecret != "" { // credentials may exist on the actual AWS node-- if so, use those. If not, override with the service key
+		os.Setenv("AWS_ACCESS_KEY_ID", accessKeyID)
+		os.Setenv("AWS_SECRET_ACCESS_KEY", accessKeySecret)
+	}
+	c := &aws.Config{
+		Region:      aws.String(region),
+		Credentials: credentials.NewEnvCredentials(),
+	}
+	s := session.Must(session.NewSession(c))
+	s3Svc := s3.New(s)
+	downloader := s3manager.NewDownloaderWithClient(s3Svc)
+
+	t := time.Now().Add(time.Duration(-60) * time.Minute)
+	ls := &s3.ListObjectsInput{
+		Bucket: aws.String(bucket),
+		Prefix: aws.String(prefix + "/" + projectID + "." + t.Format("2006-01-02-15")),
+	}
+	lso, err := s3Svc.ListObjects(ls)
+	if err != nil {
+		log.Printf(err.Error())
+	}
+	var keys []*string
+	log.Printf("GETTING LSO CONTNETS")
```

Comment:
Please don't leave debugging statements in code, or at bare minimum guard them with `klog.V(10)` or such

Reply:
Thanks